### DISCUSSION TITLE
Elementor with headless wp?(Gatsbyjs)

### DISCUSSION THREAD

**Discussion by K-Kit** [elementor/elementor#8434] [Developers Community]
Does anyone know of a way to load elementor blocks via API into a static site gen?

I'm new to wordpress and have tried adding the blocks hook to add blocks to API and am struggling to see how this can be done (if it can be).

**Comment by K-Kit** [comment]
No one else trying this?  getting pretty close to a solution but a lot of the css just isn't loading right
https://5d16eac189e8f40007ec600c--getfoundquicker.netlify.com/home-new

compared to 
https://www.getfoundquick.com/home-new

**Reply by oseisaac** [reply]
Did you eventually finish it ?

**Comment by joffff** [comment]
@K-Kit I'm interested in this approach (yet to start investigating a solution) so subscribing to this.  

Have you tried importing Elementor's JS modules into your front end?

**Comment by troydean** [comment]
I'm subscribing here too as I'm very interested in this.

**Comment by K-Kit** [comment]
Hello all,  I ended up calling it on the site since there was too much work but it did seem like a simple elementor page should work with the js imported, although that does hurt the benefits of a SSG (i think not really an expert).  There were too many other factors, and things like template sections didn't play nice.  The best thing I found before I quit was WP-static if yall are still looking for something.

I'd imagine you could edit the code from that to add some fancy image preprocessing if you wanted to.

**Comment by Kpudlo** [comment]
I've just started looking into the concept of headless myself...really interested in seeing how this goes!

**Comment by mirafzal** [comment]
https://m.youtube.com/watch?v=N5UtB36x_O8

**Reply by mikenk2010** [reply]
This is good stuff, haven't tested this solution, but wondering what will happen with hook and filter, will it append code into the Elementor

**Comment by Mofect** [comment]
No, I haven't finished it yet.

**Comment by Stelikas** [comment]
I've been trying to use WP elementor as a headless CMS builder for a web platform, although i have made it work through the API with the CSS styles and the HTML renders just find, the only issue is, most of the widgets have JS dependencies, i cannot find a way to get those dependencies in result that some widgets do not work as intended, if anyone has any idea on this it would be really helpful.
